### PR TITLE
Support recovery from in the middle of a rename.

### PR DESCRIPTION
**Pull request by brendandburns**
Code written @ 12:30am on a plane.  Treat with skepticisim...

(and don't merge yet, I want to add tests)

### CODE REVIEW

**Comment by brendandburns** [comment]
Ok, I added tests to validate this.  Please review.

thanks!
--brendan

**Comment by jlowdermilk** [comment]
lgtm.

File: pkg/kubectl/cmd/rollingupdate.go [hunk 1]
```diff
@@ -161,7 +171,7 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 
 		if len(args) >= 2 {
 			newName = args[1]
-		} else {
+		} else if oldRc != nil {
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: Wait, why would oldRc ever be nil here? If we didn't find it, wouldn't we would have returned already inside the block starting at 137?

Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: Might be time to break this function up. Too hard to follow the logic if you have to scroll thru multiple screen lengths.

**Comment by brendandburns** [comment]
Addressed comments, and merged in retries changes.  Please take another look.

Thanks!
--brendan

File: pkg/kubectl/cmd/rollingupdate.go [hunk 2]
```diff
@@ -130,16 +122,26 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
...
+		// We're in the middle of a rename, look for an RC with a source annotation of oldName
+		newRc, err := kubectl.FindSourceController(updaterClient, cmdNamespace, oldName)
+		if err != nil {
+			return err
+		}
+		return kubectl.Rename(kubectl.NewRollingUpdaterClient(client), newRc, oldName)
 	}
 
-	keepOldName := false
+	keepOldName := len(args) == 1
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: We should only be doing rename if `--image` was set. Rolling updates to an rc from filename should use the name from the schema file.

**Comment by jlowdermilk** [comment]
Some function references need fixing. From travis:
```
# github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/rollingupdate_test.go:174: undefined: addDeploymentKeyToReplicationController
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/rollingupdate_test.go:264: undefined: updateWithRetries
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd [build failed]
```

File: pkg/kubectl/cmd/rollingupdate.go (hunk 2 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/rollingupdate.go]: well, if len(args) == 1 and --image isn't set, it will error out earlier in validateArgs, but I'm happy to add an explicit check here if you prefer.

**Comment by brendandburns** [comment]
Ok, I think I forgot to push some pieces.  Re-pushed, please take another look.

File: pkg/kubectl/cmd/rollingupdate.go (hunk 2 above)
Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: I don't see that check in validateArguments. Also, `len(args) == 1` is expected for the `--filename` case, so that check wouldn't quite work. Just moving this assignment under the `if len(image) != 0` block should suffice.

**Comment by jlowdermilk** [comment]
lgtm. will merge on green.
